### PR TITLE
MenuToolBar

### PR DESCRIPTION
**Pull request by nanopay-arthur**
Relating to https://github.com/nanoPayinc/NANOPAY/pull/9463

MenuToolBar: Displays a toolbar of menus based on given menu classification (menu id prefix) (ie. sme.menu), parent (ie. sme), and title (ie. Add Contact)

### CODE REVIEW

File: src/foam/nanos/menu/MenuToolBar.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
+foam.CLASS({
+    package: 'foam.nanos.menu',
+    name: 'MenuToolBar',
+    extends: 'foam.u2.View',
+  
+    documentation: 'Creates a toolbar out menus from the given menu classification.',
```
Comment (kgrgreer) [inline src/foam/nanos/menu/MenuToolBar.js]: Grammar seems a little off.

File: src/foam/u2/ModalWrapper.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'ModalWrapper',
+  extends: 'foam.u2.View',
+  
+  documentation: `Wraps a given view with a given wrapper.`,
```
Comment (kgrgreer) [inline src/foam/u2/ModalWrapper.js]: Can you provide more docs on where and how this is used?

File: src/foam/u2/ModalWrapper.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      class: 'foam.u2.ViewSpec',
+      name: 'wrapper',
+      type: 'foam.lib.json.UnknownFObject',
+      fromJSON: function fromJSON(value, ctx, prop, json) { return value; }
+    },
+    {
+      class: 'FObjectProperty',
+      name: 'wrapperModel_',
+      factory: function() {
+        return (foam.lookup(this.wrapper.class)).create(this.wrapper, this);
```
Comment (kgrgreer) [inline src/foam/u2/ModalWrapper.js]: Extra parens not required.

File: src/foam/nanos/menu/MenuToolBar.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/nanos/menu/MenuToolBar.js:7]: Add copyright header.

File: src/foam/nanos/menu/MenuToolBar.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    package: 'foam.nanos.menu',
+    name: 'MenuToolBar',
+    extends: 'foam.u2.View',
+  
+    documentation: 'A toolbar made out of menus according to a given classification.',
```
Comment (kgrgreer) [inline src/foam/nanos/menu/MenuToolBar.js]: What is meant by 'classification'?

File: src/foam/u2/ModalWrapper.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/u2/ModalWrapper.js]: Add copyright header.

Comment (kgrgreer) [inline src/foam/u2/ModalWrapper.js]: How is this different from our current Border support?

File: src/foam/nanos/menu/MenuToolBar.js [hunk 7]
```diff
@@ -3,8 +3,8 @@ foam.CLASS({
     name: 'MenuToolBar',
     extends: 'foam.u2.View',
   
-    documentation: 'Creates a toolbar out menus from the given menu classification.',
-  
+    documentation: 'A toolbar made out of menus according to a given classification.',
+
```
Comment (kgrgreer) [inline src/foam/nanos/menu/MenuToolBar.js]: remove spaces

**Review by jlhughes — APPROVED** [review]

**Pull request merged**